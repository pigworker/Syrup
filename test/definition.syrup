not(<Bit>) -> <Bit>
not(X) = nand(X,X)

and(<Bit>, <Bit>) -> <Bit>
and(X,Y) = !nand(X,Y)

or(<Bit>, <Bit>) -> <Bit>
or(X,Y) = nand(!X,!Y)

1 -> <Bit>
1 = !0

definition f [X,Y,Z] [0,0,0,0,1,0,0,1]
definition g [X,Y,X] [0,0,0,0,1,0,0,1]
definition h [] []
definition i [X] [0]
definition j [] [0]