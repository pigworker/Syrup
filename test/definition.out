Circuits not, and, and or are defined.

DNF circuit for f:
f(<Bit>, <Bit>, <Bit>) -> <Bit>
f(X, Y, Z) = X & Y & Z | X & !Y & !Z

Error: You are redefining the local variable X.

Error: Invalid truth table output for h.

Error: Invalid truth table output for i.

DNF circuit for j:
j() -> <Bit>
j() = zero()
